src/components/Plants/index.tsx

```diff
@@ -73,12 +73,18 @@ export default function PlantsDashboard(props: PlantsDashboardProps): JSX.Elemen
         )}
         <Box display='flex' alignItems='center' paddingTop={isMobile ? 2 : 0}>
           <Typography sx={{ paddingRight: 1, fontSize: '16px', fontWeight: 500 }}>{strings.PLANTING_SITE}</Typography>
-          <Select
-            options={plantingSites.map((ps) => ps?.name || '')}
-            onChange={onChangePlantingSite}
-            selectedValue={selectedPlantingSite?.name}
-            placeholder={strings.SELECT}
-          />
+          {plantingSites ? (
+            <Select
+              options={plantingSites.map((ps) => ps?.name || '')}
+              onChange={onChangePlantingSite}
+              selectedValue={selectedPlantingSite?.name}
+              placeholder={strings.SELECT}
+            />
+          ) : (
+            <Box sx={{ position: 'fixed', top: '50%', left: '50%' }}>
+              <CircularProgress />
+            </Box>
+          )}
```

Comment:
would be more readable to have this outside - also  will avoid mistakes like showing planting sites details in this case when it should't show

```
  if (!plantingSites) {
   return (
     <Box>
      spinner
   </box>
   )
  }

return (
   normal content
);

```